config/variables.go

```diff
@@ -0,0 +1,161 @@
+package config
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/boilerplate/util"
+	"github.com/gruntwork-io/boilerplate/errors"
+	"strings"
+	"io/ioutil"
+	"gopkg.in/yaml.v2"
+)
+
+// Get a value for each of the variables specified in boilerplateConfig. The value can come from the user (if the
+// non-interactive option isn't set), the default value in the default value in the config, or a command line option.
+func GetVariables(options *BoilerplateOptions, boilerplateConfig *BoilerplateConfig) (map[string]string, error) {
+	variables := map[string]string{}
+
+	for _, variable := range boilerplateConfig.Variables {
+		value, err := getVariable(variable, options)
+		if err != nil {
+			return variables, err
+		}
+		variables[variable.Name] = value
+	}
+
+	return variables, nil
+}
+
+// Get a value for the given variable. The value can come from the user (if the non-interactive option isn't set), the
+// default value in the default value in the config, or a command line option.
+func getVariable(variable Variable, options *BoilerplateOptions) (string, error) {
+	valueFromVars, valueSpecifiedInVars := getVariableFromVars(variable, options)
+
+	if valueSpecifiedInVars {
+		util.Logger.Printf("Using value specified via command line options for variable '%s': %s", variable.Name, valueFromVars)
+		return valueFromVars, nil
+	} else if options.NonInteractive && variable.Default != "" {
+		// TODO: how to disambiguate between a default not being specified and a default set to an empty string?
```

Comment:
We could have the convention for the `yml` file that if `default:` is specified with a blank value, to record an empty-value escape-sequence like `##EMPTY-VAL##`. But yes, what if someone actually uses that string, so maybe better to do nothing for now until this is an issue.